packages/flutter/lib/src/material/carousel.dart

```diff
@@ -1616,7 +1739,9 @@ class CarouselController extends ScrollController {
     }
 
     final bool hasFlexWeights = _carouselState!._flexWeights?.isNotEmpty ?? false;
-    index = index.clamp(0, _carouselState!.widget.children.length - 1);
+    final int itemCount =
```

Comment:
If we use the lazy loading constructors, and if widget.itemCount is not specified, itemCount will default to 0 and then index.clamp(0, itemCount - 1) will have an invalid upper limit.